src/main/scala/IsRetweetClassifier.scala

```diff
@@ -0,0 +1,84 @@
+import org.apache.spark.ml.Pipeline
+import org.apache.spark.ml.classification.LogisticRegression
+import org.apache.spark.sql.types._
+import org.apache.spark.sql.{Row, SparkSession}
+import org.apache.spark.ml.feature.{HashingTF, Tokenizer}
+import org.apache.spark.sql.functions._
+
+object IsRetweetClassifier {
+  def main(args: Array[String]) {
+    val spark = SparkSession
+      .builder()
+      .appName("Pipeline")
+      .master("yarn")
+      .getOrCreate()
+
+    import spark.sqlContext.implicits._
+
+    val text = spark
+      .sparkContext.wholeTextFiles("hdfs:///user/ilos/tweets/tweets.txt")
+      .map { case (filename, content) => content}
+
+    val tweets = text
+      .flatMap(line => line.split("List\\("))
+
+    val nonEmptyTweets = tweets
+      .map(line => line.stripLineEnd)
+      .filter(line => line.length() > 0)
+
+    val nonEmptyTweetsNormalized = nonEmptyTweets
+      .map(line => line.replaceAll(", true\\)", ", 1"))
+      .map(line => line.replaceAll(", false\\)", ", 0"))
+      .map(line => line.replaceAll(", und, ", ", ??, "))
+
+    val structuredDataRDD = nonEmptyTweetsNormalized.map(line => Row(
+      line.substring(0, line.length() - 7),
+      line.substring(line.length() - 5, line.length() - 3),
+      line.substring(line.length() - 1).toInt
+    ))
+
+    val schema = new StructType()
+      .add(StructField("text", StringType, nullable = false))
+      .add(StructField("language", StringType, nullable = false))
+      .add(StructField("label", IntegerType, nullable = false))
+
+    val structuredData = spark
+      .createDataFrame(structuredDataRDD, schema)
+      .withColumn("id", monotonically_increasing_id())
+
+    val trainLength = (structuredData.count() * 0.8).toInt
+    val testLength = (structuredData.count() - trainLength).toInt
+
+    val train = structuredData.limit(trainLength)
```

Comment:
Yes, it's a bug. In line 'val model = pipeline.fit(structuredData)' I should use train instead of structuredData. So, I just fit my model on both datasets. I will fix it.